saleor/graphql/order/mutations/draft_orders.py

```diff
@@ -225,3 +206,118 @@ def mutate(cls, root, info, id):
             type=OrderEvents.PLACED_FROM_DRAFT.value,
             user=info.context.user)
         return DraftOrderComplete(order=order)
+
+
+class DraftOrderLineAdd(BaseMutation):
+    order = graphene.Field(Order, description='A related draft order.')
+    order_line = graphene.Field(OrderLine, description='A newly created order line.')
+
+    class Arguments:
+        id = graphene.ID(
+            required=True,
+            description='ID of the draft order to add the lines to.')
+        input = OrderLineCreateInput(
+            required=True,
+            description="""
+            Variant line input consisting of variant ID and quantity of
+            products.""")
+
+    class Meta:
+        description = 'Add order line to a draft order.'
+
+    @classmethod
+    @permission_required('order.manage_orders')
+    def mutate(cls, root, info, id, input):
+        errors = []
+        order = cls.get_node_or_error(info, id, errors, 'id', Order)
+        variant_id = input['variant_id']
+        variant = cls.get_node_or_error(
+            info, variant_id, errors, 'lines', ProductVariant)
+
+        if not (order or variant):
+            return DraftOrderLineAdd(errors=errors)
+
+        if order.status != OrderStatus.DRAFT:
+            cls.add_error(
+                errors, 'order_id', 'Only draft orders can be edited.')
+
+        line = None
+        if not errors:
+            line = add_variant_to_order(
+                order, variant, input['quantity'], allow_overselling=True)
```

Comment:
I think we should check if quantity is larger than 0

Reply:
True. I'll add a manual check but it would be very helpful if we were able to use Django validators for that purpose. We'll have to think about something like that in our `BaseMutation`.

Reply:
Related: https://github.com/mirumee/saleor/issues/2912
At this moment 0 is allowed value on the quantity field in the models.